### PR TITLE
New data structure proposal

### PR DESCRIPTION
**Pull request by exu**
## Pull request description 

Change API objects to be more concise and easier to understand
The idea is to get rid of 
```
script-executions
\- execution 
      \-- result
```

 and have executions with results only instead (in API server)

I think it should be easier to uderstand whats going on  

Second idea is to have single ID passed to executor (ID generated by API server) 
* current solution has 2 IDs one generated by API server second one by executor (what confuse most of people who looks at it :D) 

Below old and new data structures: 
``` 
{
	"id": "614b8b07ebfab223ca07d9f8",
	"scriptType": "postman/collection",
	"name": "wholly-equal-humpback",
	"execution": {
		"id": "614b8b07ba01d5074890d43a",
		"scriptContent": "CONTENT",
		"status": "success",
		"result": {
			"status": "success",
			"output": "EXECUTION RESULT TERMINAL BASED OUTPUT",
			"outputType": "text/plain",
			"startTime": "2021-09-22T19:59:06.113Z",
			"endTime": "2021-09-22T19:59:06.365Z",
			"steps": [
				{
					"name": "Health",
					"duration": "201ms",
					"status": "success",
					"assertionResults": [
						{
							"name": "Status code is 200",
							"status": "success"
						}
					]
				}
			]
		},
		"startTime": "2021-09-22T19:59:05.043Z",
		"endTime": "2021-09-22T19:59:06.384Z"
	}
}
```

new one (proposal)
```
{
	"id": "614b8b07ebfab223ca07d9f8",
	"scriptType": "postman/collection",
	"scriptContent": "CONTENT",
	"name": "wholly-equal-humpback",
	"result": {
		"status": "success",
		"output": "EXECUTION RESULT TERMINAL BASED OUTPUT",
		"outputType": "text/plain",
		"startTime": "2021-09-22T19:59:06.113Z",
		"endTime": "2021-09-22T19:59:06.365Z",
		"steps": [
			{
				"name": "Health",
				"duration": "201ms",
				"status": "success",
				"assertionResults": [
					{
						"name": "Status code is 200",
						"status": "success"
					}
				]
			}
		]
	},
}
```
what would need to be done to have such structure ?  (OpenAPI spec) 
 
## Checklist (choose whats happened)

- [X] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

## Breaking changes

- ScriptExecution -> Execution
- Execution -> Result
- Flatten ExecutionResult to Result
- Remove additional fields

- Refactor code
- REST based executors - on script execution should get id passed from API server and should manage this ID internally - (POST /executions/{ID} ? or id in body ? ) 
- they need to be able to GET /executions/{ID}) ID will be same for API - Execution and REST executor internal ID (now it’s different as it’s separate storages - what is confusing)  

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by jasmingacic — APPROVED** [review]
This makes sense and makes data structures much leaner.

File: api/v1/kubtest.yaml [hunk 1]
```diff
@@ -442,27 +448,10 @@ components:
           description: 'test execution end time'
           format: date-time
 
-    Execution:
+    Result:
```
Comment (olensmar) [inline api/v1/kubtest.yaml:451]: lets call this ExecutionResult (there could be other types of results down the line..)

**Review by nicufk — APPROVED** [review]
Looks pretty good, Good Job!

File: api/v1/kubtest.yaml (hunk 1 above)
Comment (exu) [reply api/v1/kubtest.yaml:451]: ok makes sense gimme 5 mins :D

**Pull request merged**